src/HttpClient.php

```diff
@@ -0,0 +1,103 @@
+<?php
+declare(strict_types = 1);
+
+namespace jsamhall\ShipEngine;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+
+/**
+ * Class HttpClient
+ * @package jsamhall\ShipEngine
+ */
+class HttpClient extends Client
+{
+    /** @var int */
+    protected $lastStatus;
+
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+    }
+
+    # region Functions
+
+    public function postJson(string $route, array $payload, string $key = null)
+    {
+        return $this->tryCatchRequest('POST', $route, $payload, $key);
+    }
+
+    public function getJson(string $route, array $params = [], string $key = null)
+    {
+        return $this->tryCatchRequest('GET', $route, $params, $key);
+    }
+
+    public function putJson(string $route, array $payload = [], string $key = null)
+    {
+        return $this->tryCatchRequest('PUT', $route, $payload, $key);
+    }
+
+    public function deleteJson(string $route, array $params = []): bool
+    {
+        $this->tryCatchRequest('DELETE', $route, $params);
+        return $this->isSuccessful();
+    }
+
+    public function isSuccessful(): bool
+    {
+        $successful = [200, 201, 202, 203, 204];
+
+        return in_array($this->lastStatus, $successful);
+    }
+
+    private function getResponseData(array $data, $key = null)
```

Comment:
Should this be with the rest of the private functions?

Reply:
yep, moved